tests/test_api.py

```diff
@@ -69,6 +69,60 @@ def test_get_version(self):
         res = self.assertGetResponse("/version", requests.codes.ok, headers=get_auth_header())
         self.assertEquals(res.json['version_info']['version'], os.environ['DSS_VERSION'])
 
+    @testmode.standalone
+    def test_put_file_pattern(self):
+        """
+        Tests the regex pattern filtering on the PUT file/{uuid} endpoint.
+
+        Ensures that paths with leading slashes (absolute paths) and
+        relative path shortcuts (".", "~/", and "..") are disallowed.
+        """
+
+        examples_of_bad_paths = ['.', '..',
+                                 '~/path',
+                                 './', '../', './path', './../path', '../path', '../../path',
+                                 'path/../path', 'path/..',
+                                 '/path', '/path.json', '/bad..path']
+
+        examples_of_good_paths = ['path', 'path.json', 'good..path', 'path.json/', 'good..path/',
+                                  'pa/th.json', 'go/od..path', 'a/b/c/d', 'a/b.c/d', '.bashrc']
+
+        for path, replica in [("files", "aws"), ("files", "gcp")]:
```

Comment:
Only test one replica. The replica does not affect the outcome.

Reply:
Alright.